plugin/core/transports.py

```diff
@@ -215,7 +215,7 @@ def read_stdout(self) -> None:
                         content_length = 0
                     state = STATE_HEADERS
 
-            except IOError as err:
+            except (AttributeError, IOError) as err:
```

Comment:
I'd like a different fix for this - there is no clarification why AttributeError is in there and if it's only the tests that trigger it then the fix should be in test code.

Reply:
After thinking about it for a while the process may also become None in production code. Who makes sure that the process is still alive after checking wether it is alive?

The only way to handle this is by "just reading from it", and using exception mechanism to handle sudden failure.

It's kind of the same situation with reading files. You can check wether a file is present on the filesystem, but to really make sure that the file exists you have to open it and handle an exception if opening the file fails.

Reply:
You are right. It would be nice to avoid catching _all_ AttributeErrors in this code. I thought about adding a getter to `self.process.stdout` but the point is we _do_ want the same exception handling logic anyways.

If there is no obvious way to only catch the missing stdout then let's go ahead with this.

Reply:
This error still feels really broad to catch here.

 I think we can declare a `UnexpectedProcessExitError` and do something like this:
```python
def safe_stdout(self) -> 'IO[Any]':
  if self.process:
    return self.process.stdout
  else:
    raise UnexpectedProcessExitError()
```

Reply:
I'll see what I can do :)